tools/@aws-cdk/eslint-plugin/test/rules/fixtures/no-invalid-path/variable-path.ts

```diff
@@ -0,0 +1,4 @@
+import * as path from 'path';
+
+const myPath = [__dirname, '..', '../..', 'fixtures'];
+path.join(...myPath); // Should fail but does not currently because myPath is treated as one 'SpreadElement' argument.
```

Comment:
What shall we do here? Disallow it? What about a situation like

```ts
export function packageJsonDir(dir: string) {
  fs.readFileSync(path.join(dir, 'package.json'));
}
```

I'm thinking about this too, just ran out of time today.

Reply:
Interesting. I think we can flag ANY array or sequence of arguments in which `__dirname` is followed by strings as suspicious. Doesn't even matter if it's in a `path.join`.

Does this actually occur? Or are you being safe?

As for the `packageJsonDir` -- well those are the limits of static checking, aren't they? We're not trying to catch everything, we're just trying to catch common thoughtless mistakes. How likely is it that people would actually write  that?